source/includes/code-examples/insert-many/InsertMany.cs

```diff
@@ -0,0 +1,113 @@
+using MongoDB.Bson;
+using MongoDB.Bson.Serialization.Attributes;
+using MongoDB.Bson.Serialization.Conventions;
+using MongoDB.Driver;
+using static System.Console;
+
+namespace CsharpExamples.UsageExamples;
+
+public class InsertMany
+{
+    private static IMongoCollection<Restaurant> _restaurantsCollection;
+    private static string _mongoConnectionString = "<Your MongoDB URI>";
+
+    public static void Main(string[] args)
+    {
+        Setup();
+
+        // Attempt to find document before insert
+        var filter = Builders<Restaurant>.Filter.Eq("name", "Mongo's Pizza");
+
+        var foundRestaurants = _restaurantsCollection.Find(filter).ToList();
+
+        WriteLine($"Number of restaurants found before insert: {foundRestaurants.Count}");
+
+        // Extra space for console readability
+        WriteLine();
+
+        WriteLine("Inserting documents...");
+        InsertManyRestaurants();
+
+        // find and print newly inserted document
+        foundRestaurants = _restaurantsCollection.Find(filter).ToList();
+
+        WriteLine($"Number of restaurants inserted: {foundRestaurants.Count}");
+
+        Cleanup();
+    }
+
+    private static void InsertManyRestaurants()
+    {
+        // delete sample document if already exists
+        Cleanup();
+
+        // start-insert-many
+        // Helper method to generate 5 new restaurants
+        List<Restaurant> restaurants = GenerateDocuments();
+
+        _restaurantsCollection.InsertMany(restaurants);
+        // end-insert-many
+    }
+
+    private static void Setup()
+    {
+        // This allows automapping of the camelCase database fields to our models. 
+        var camelCaseConvention = new ConventionPack { new CamelCaseElementNameConvention() };
+        ConventionRegistry.Register("CamelCase", camelCaseConvention, type => true);
+
+        // Establish the connection to MongoDB and get the restaurants database
+        var mongoClient = new MongoClient(_mongoConnectionString);
+        var restaurantsDatabase = mongoClient.GetDatabase("sample_restaurants");
+        _restaurantsCollection = restaurantsDatabase.GetCollection<Restaurant>("restaurants");
+    }
+
+    private static List<Restaurant> GenerateDocuments()
+    {
+        // Generate 5 new restaurant documents
+        List<Restaurant> restaurantsList = new List<Restaurant>();
+        for (int i = 1; i <= 5; i++)
+        {
+            Restaurant newRestaurant = new()
+            {
+                Name = "Mongo's Pizza",
+                RestaurantId = $"12345-{i}",
```

Comment:
This collection has both, I used RestaurantId since it's not an auto-generated field like _id (which translates to Id here), so I think it's a bit safer to work with. I don't want to encourage users to modify the _id, or to do anything with it if it's not necessary